[chongsun2002] iP

chongsun2002 opened the pull request:
### Yap
> If you don't even know what you need to do, how can you do what you need to do?

Yap is a task manager which helps you track the things that you need to do and free up your mind for other tasks💯!

- _Easy_ to use
- Text and GUI based
- ~~EXPENSIVE~~ **FREE**

It currently possesses the following capabilities:

- [ ] Managing tasks
- [ ] Addition of tasks with deadlines and events
- [ ] Sending you reminders (WIP)

## How can I use it?

1. Download the jar file [here](https://github.com/chongsun2002/ip/releases/tag/A-Jar)
2. Run it
3. Add your tasks

If you are curious, this is how it scans in your queries:
```Java
Storage storage = new Storage("yap.txt", "./data/");
TaskList taskList = new TaskList(storage);
Ui ui = new Ui(storage, taskList);

while (true) {
    String userInput = scanner.nextLine();
    try {
        int userInputParseStatus = ui.reactToUserInput(userInput);
        if (userInputParseStatus == 0) {
            break;
        }
    } catch (InputException | DateTimeParseException exception) {
        System.out.println(exception.getMessage());
        System.out.println(SEPARATOR);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

chongsun2002 committed:
Change name to Yap and customise messages

chongsun2002 committed:
Echo user query until bye

Chatbot is saying bye before taking in user input\nUser cannot input anything\nInfinite loop added that scans for input\n

chongsun2002 committed:
Add ability to store text, list function

Chatbot just echoing user input
Gave chatbot functionality by allowing it to store user inputs
Allow chatbot to list out stored inputs

chongsun2002 committed:
add Task class and mark as done functionality

Tasks has no status and are just a string.
Add Task class for functionality.
Allow tasks to be marked as done or not done.

chongsun2002 committed:
Add support for multiple task type

All task types the same

Add Deadline, Event, Todo classes

Add InputParser class to abstract code from Yap class

chongsun2002 committed:
Write expected output for testcase, tested code

chongsun2002 committed:
Add exceptions to handle erroneous inputs

Erronous input causes program to crash

Add InputException class, superclass for all input exceptions

Add DeadlineException and EventException which inheirt from InputException

chongsun2002 committed:
Add delete feature, change tasklist data structure

Tasklist not resizable, change from array to arraylist

chongsun2002 committed:
Add storage feature, more exceptions for bad storage

Yap does not have a storage feature, resulting in all data being loss when program is stopped

Storage feature is necessary as users will want tasks to be saved even when they exit the program

Create such storage feature which saves the task as Task Type | Completion Status | Duration related stuff

chongsun2002 committed:
Add date support for event and deadline classes

suhayl13 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString()
+                + String.format(" (by: %s)", deadline.format(DateTimeFormatter.ofPattern("MMM d yyyy")));
```
Good job placing line breaks to improve readability! Maybe have the same indentation for "deadline.format..." as well?

suhayl13 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,37 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
```
Great work grouping related packages together!

suhayl13 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,76 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
```
Perhaps group according to their relevant packages?

suhayl13 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,76 @@
+        if (!file.exists()) {
+            try {
+                if (!file.createNewFile()) {
+                    System.out.println("The file did not exist and there was a problem creating it.");
+                }
+            } catch (IOException exception) {
+                System.out.println("The file did not exist and there was a problem creating it!");
```
Could this be refactored as the same message is outputted for both scenarios?

suhayl13 reviewed (commented):
LGTM 👍. Just a few nits to pick.

oneBoz commented on src/main/java/InputParser.java:
```diff
@@ -0,0 +1,46 @@
+        return new ToDo(taskName);
+    }
+    public static Event parseInputAsEvent(String input) throws InputException {
+        String[] parts, times;
```
Perhaps timeRange would better reflect that the array holds both the start time and the end times of the event.

oneBoz commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,76 @@
+import java.io.File;
+import java.io.FileNotFoundException;
```
Perhaps separate the packages via a white line?

oneBoz commented on src/main/java/InputParser.java:
```diff
@@ -0,0 +1,46 @@
+            throw new EventException(EventException.EventExceptionType.NOTO);
+        }
+        try {
+            description = parts[0].replace("event ", "").trim();
```
I like the names of the temporary variables

oneBoz commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,92 @@
+            String description;
+            boolean isDone;
+            switch (taskDetails[0].charAt(0)) {
+            case 'T':
```
I like how you managed to stick to the correct switch format

oneBoz reviewed (commented):
Overall LGTM! Just some minor suggestions!

chongsun2002 committed:
Add Ui class for better OOP

chongsun2002 committed:
Move classes into packages storage, task and ui

chongsun2002 committed:
Merge branch 'add-gradle-support'

chongsun2002 committed:
Add gradle support

chongsun2002 committed:
Add junit testcase for TaskList and Parser

chongsun2002 committed:
Add find functionality

chongsun2002 committed:
Fix coding standard violations, reorder imports

chongsun2002 committed:
Resolve merge conflict by removing import

chongsun2002 committed:
Fix bug from merge conflict resolution

chongsun2002 committed:
Add JavaDoc for methods in yap.task and yap.storage.

chongsun2002 committed:
Merge branch 'branch-A-JavaDoc'

chongsun2002 committed:
Merge all branches into master branch, make minor bug fixes

chongsun2002 committed:
Add checkstyle and suppressions files to checkstyle with gradle

chongsun2002 committed:
Fix bug with tasklist due to typo

chongsun2002 committed:
Merge branch 'branch-A-CheckStyle'

chongsun2002 committed:
Fix all violations found by checkstyle

chongsun2002 committed:
Fix weak slap in parser class

chongsun2002 committed:
Add GUI based on tutorial

chongsun2002 committed:
Add assertions to command class

Current command class does not verify input before trying to parse it

This will cause an error to be thrown if the user inputs something wrongly

Insert an assert line in every command to verify it is a correct command before parsing

Assertions are a good way to check a variable's state at a current point of the program

chongsun2002 committed:
Merge pull request #2 from chongsun2002/branch-A-Assertions

Add assertions to Yap

chongsun2002 committed:
Change image file in mainwindow

The GUI is not loading properly due to wrong filepath

Let's change the filename so there is no error loading

chongsun2002 committed:
Fix code quality violations

Codebase has some style violations

Code should look like a single person wrote it

Let's run checkstyle and look through the codebase

This is the easiest way to detect violations

chongsun2002 committed:
Merge branch 'master' into branch-A-CodeQuality
Other changes pushed into master branch resulted
in this branch being behind.

Let's merge the master branch into this branch
to sync them up.

chongsun2002 committed:
Update Command and UI to follow SRP

The UI and Command prints out the results of the command.
This is not ideal as the GUI would not work properly.

Let's change the methods in these classes to return strings of the
outputs.

This will allow the output to be easily passed to the GUI while
maintaining usability of the Yap chatbot on command line.

chongsun2002 committed:
Add support for fixed duration tasks

chongsun2002 committed:
Add Ui.png and help command

chongsun2002 committed:
Update README.md with user guide
